e2e/testsuite/testsuite.go

```diff
@@ -177,52 +177,60 @@ func (s *E2ETestSuite) SetupTest() {
 	s.SetupPath(ibc.DefaultClientOpts(), defaultChannelOpts(s.GetAllChains()))
 }
 
-// SetupPath creates a path between the chains using the provided client and channel options.
+// SetupPath creates paths between the all chains using the provided client and channel options.
 func (s *E2ETestSuite) SetupPath(clientOpts ibc.CreateClientOptions, channelOpts ibc.CreateChannelOptions) {
 	s.T().Logf("Setting up path for: %s", s.T().Name())
-	r := s.relayer
 
 	ctx := context.TODO()
 	allChains := s.GetAllChains()
 	for i := 0; i < len(allChains)-1; i++ {
 		chainA, chainB := allChains[i], allChains[i+1]
-		pathName := s.generatePathName()
-		s.T().Logf("establishing path between %s and %s on path %s", chainA.Config().ChainID, chainB.Config().ChainID, pathName)
+		_, _ = s.CreateNewPath(ctx, chainA, chainB, clientOpts, channelOpts)
+	}
+}
+
+func (s *E2ETestSuite) CreateNewPath(ctx context.Context, chainA ibc.Chain, chainB ibc.Chain, clientOpts ibc.CreateClientOptions, channelOpts ibc.CreateChannelOptions) (ibc.ChannelOutput, ibc.ChannelOutput) {
```

Comment:
```suggestion
func (s *E2ETestSuite) CreateNewPath(ctx context.Context, chainA, chainB ibc.Chain, clientOpts ibc.CreateClientOptions, channelOpts ibc.CreateChannelOptions) (ibc.ChannelOutput, ibc.ChannelOutput) {
```

not feeling too strong about this, but I think it makes the signature a bit easier to read.

Reply:
I am OK with both options. I am undecided on which approach I usually prefer. I guess it depends on the signature. In this case I think I agree that it would make it a bit less verbose since it anyway has so many arguments.